vendors/infineon/boards/xmc4800_plus_optiga_trust_x/aws_tests/application_code/main.c

```diff
@@ -0,0 +1,305 @@
+#include "iot_system_init.h"
+#include "iot_logging_task.h"
+#include "iot_wifi.h"
+#include "aws_clientcredential.h"
+#include "aws_dev_mode_key_provisioning.h"
+
+/* Logging Task Defines. */
+#define mainLOGGING_MESSAGE_QUEUE_LENGTH    ( 15 )
+#define mainLOGGING_TASK_STACK_SIZE         ( configMINIMAL_STACK_SIZE * 8 )
+
+/* Unit test defines. */
+#define mainTEST_RUNNER_TASK_STACK_SIZE     ( configMINIMAL_STACK_SIZE * 16 )
+
+/* The task delay for allowing the lower priority logging task to print out Wi-Fi 
+ * failure status before blocking indefinitely. */
+#define mainLOGGING_WIFI_STATUS_DELAY       pdMS_TO_TICKS( 1000 )
+
+/* The name of the devices for xApplicationDNSQueryHook. */
+#define mainDEVICE_NICK_NAME                "XMC4800_IOTKIT_TEST"
+
+/**
+ * @brief Application task startup hook for applications using Wi-Fi. If you are not 
+ * using Wi-Fi, then start network dependent applications in the vApplicationIPNetorkEventHook
+ * function. If you are not using Wi-Fi, this hook can be disabled by setting 
+ * configUSE_DAEMON_TASK_STARTUP_HOOK to 0.
+ */
+void vApplicationDaemonTaskStartupHook( void );
+
+/**
+ * @brief Connects to Wi-Fi.
+ */
+static void prvWifiConnect( void );
+
+/**
+ * @brief Initializes the board.
+ */
+static void prvMiscInitialization( void );
+
+/* Ensure the FreeRTOS heap is not crossing SRAM boundaries */
+__attribute__((section("ETH_RAM"))) uint8_t ucHeap[ configTOTAL_HEAP_SIZE ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Application runtime entry point.
+ */
+int main( void )
+{
+    /* Perform any hardware initialization that does not require the RTOS to be
+     * running.  */
+    prvMiscInitialization();
+
+    /* Create tasks that are not dependent on the Wi-Fi being initialized. */
+    xLoggingTaskInitialize( mainLOGGING_TASK_STACK_SIZE,
+                            tskIDLE_PRIORITY,
+                            mainLOGGING_MESSAGE_QUEUE_LENGTH );
+
+    /* FIX ME: If you are using Ethernet network connections and the FreeRTOS+TCP stack,
+     * uncomment the initialization function, FreeRTOS_IPInit(), below. */
+    /*FreeRTOS_IPInit( ucIPAddress,
+     *                 ucNetMask,
+     *                 ucGatewayAddress,
+     *                 ucDNSServerAddress,
+     *                 ucMACAddress );
+     */
+
+    /* Start the scheduler.  Initialization that requires the OS to be running,
+     * including the Wi-Fi initialization, is performed in the RTOS daemon task
+     * startup hook. */
+    vTaskStartScheduler();
+
+    return 0;
+}
+/*-----------------------------------------------------------*/
+
+static void prvMiscInitialization( void )
+{
+	CONSOLE_IO_Init();
+}
+/*-----------------------------------------------------------*/
+
+void vApplicationDaemonTaskStartupHook( void )
+{
+    /* FIX ME: Perform any hardware initialization, that require the RTOS to be
+     * running, here. */
+    //vDevModeKeyProvisioning();
```

Comment:
Similar to above, as long as the PKCS #11 tests pass on OPTIGA Trust X without any of the key and certificate importing stuff that this call is meant for, this block could be removed. Probably less confusing that way.

Reply:
Agree.